test/tests.bats

```diff
@@ -1212,6 +1215,11 @@ load 'test_helper/bats-assert/load'
 
 
 @test "checking user login: predefined user can login" {
+  # This should really not be necessary, but this test sometimes fails, probably due to timing
```

Comment:
is there a file or a socket within the container that instead could signalise that the container has properly started?

to prevent having the complete test command twice something like https://stackoverflow.com/a/12321815/4754613 could be done as well

Reply:
Unfortunately not, there are other issues/PRs about that and it seems to be non-trivial to detect when everything is up and running. If we could that could speed up all the tests, so it would be an excellent separate PR. Plus possibly the detect changes job needs to run once, not sure in this case.

I'm not a bash expert, so I tried to keep things simple. We could revert this part if you like though, the most important changes are the checksum file and moving the test files to a read-only mount.

Reply:
yes. already seen al the "sleep xx" in the scripts, but did not yet look into if this can be done any better way. I don't think you need to revert, but I would add a "TODO" so its clear that this still needs to be improved upon.

Reply:
A combination of checking the logs (write something when start-mailserver.sh is done) and connecting to a socket should work. I'll add TODOs to the sleeps I added.